### PR TITLE
Add initialValue to useRef

### PR DESCRIPTION
**Pull request by TanaseHagi**

### CODE REVIEW

File: packages/flutter_hooks/lib/src/primitives.dart [hunk 1]
```diff
@@ -4,19 +4,20 @@ part of 'hooks.dart';
...
 
 /// Creates an object that contains a single mutable property.
 ///
 /// Mutating the object's property has no effect.
 /// This is useful for sharing state accross `build` calls, without causing
 /// unnecessary rebuilds.
-ObjectRef<T> useRef<T>() {
-  return useMemoized(() => ObjectRef<T>());
+ObjectRef<T> useRef<T>(T initialValue) {
+  return useMemoized(() => ObjectRef<T>(initialValue));
```
Comment (rrousselGit) [inline packages/flutter_hooks/lib/src/primitives.dart:20]: Could you add a test for this? Thanks!

Comment (TanaseHagi) [reply packages/flutter_hooks/lib/src/primitives.dart:20]: Sure, [here](https://github.com/rrousselGit/flutter_hooks/pull/244/commits/763085d244da9dbc03070c0080c47c157844051b) you go!

**Comment by TanaseHagi** [comment]
any update?

File: packages/flutter_hooks/test/memoized_test.dart [hunk 2]
```diff
@@ -16,7 +16,7 @@ void main() {
 
     await tester.pumpWidget(
       HookBuilder(builder: (context) {
-        ref = useRef();
+        ref = useRef<int?>(null);
```
Comment (rrousselGit) [inline packages/flutter_hooks/test/memoized_test.dart:19]: These tests should be unchanged. Could you instead add a new test that checks both a null and non-null default value?

Comment (rrousselGit) [reply packages/flutter_hooks/test/memoized_test.dart:19]: And that makes sure that when the widget rebuilds, the state is not reset

Comment (TanaseHagi) [reply packages/flutter_hooks/test/memoized_test.dart:19]: Hope [this](https://github.com/rrousselGit/flutter_hooks/pull/244/commits/3ee484220696492a320762ebf5fdb59de18b8897) will suffice

**Review by rrousselGit — APPROVED** [review]

**Comment by rrousselGit** [comment]
Thanks!

**Comment by TanaseHagi** [comment]
Thank you!
